ethereum/contracts/ERC20App.sol

```diff
@@ -42,16 +35,17 @@ contract ERC20App is Application {
         public
         override
     {
-        require(_data.length == MESSAGE_LENGTH, "Message must contain 104 bytes for a successful decoding");
+        require(_data.length >= PAYLOAD_LENGTH, "Invalid Payload");
```

Comment:
Could you add a comment why we allow longer payloads? Thanks!

Reply:
This was due to uncertainty as to whether the bridge contract was correctly removing the verification data from the end of a message. The verification data is used for replay protection and so is not required by target applications. At the time, I opted to loosen restrictions in order to get everything working together.

I've cut https://github.com/Snowfork/polkadot-ethereum/issues/129 to address this issue.